README.md

```diff
@@ -110,6 +115,94 @@ You can verify your smart contract on Etherscan by running:
 yarn verify --network network_name
 ```
 
+## Hook Example
+
+- useScaffoldContract
+- useScaffoldContractRead
+- useScaffoldContractWrite
+- useScaffoldEventSubscriber
+- useScaffoldEventHistory
+- useDeployedContractInfo
+
+  **useScaffoldContract**:
+  Use to gets a deployed contract by contract name and returns a contract instance
+  Can also be use to read and write to the deployed smart contract
+
+  ```ts
+  const { data: yourContract } = useScaffoldContract({
+    contractName: "YourContract",
+  });
+  // will return the greeting and can be call in any function unlike useScaffoldContractRead
+  await yourContract.greeting();
+
+  //will be use to write to a contract and can be called in any function
+  await yourContract.setGreeting("Hello");
+  ```
+
+  **useScaffoldContractRead**:
+  Use this hook to read a value from your deployed contracts.
+
+  ```ts
+  const { data: totalCounter } = useScaffoldContractRead({
+    contractName: "YourContract",
+    functionName: "getGreeting",
+    // if it does not accept any args it can be optional
+    args: ["ARGUMENTS IF THE FUNCTION ACCEPTS ANY"],
+  });
+  ```
+
+  **useScaffoldContractWrite**:
+  Use this hook to write to your deployed contracts.
+
+  ```ts
+  const { writeAsync, isLoading } = useScaffoldContractRead({
+    contractName: "YourContract",
+    functionName: "setGreeting",
+    args: ["The value to set"],
+    //value if the function is payable and sends eth to it
+    value: "0.01",
+  });
+  ```
+
+  **useScaffoldEventSubscriber**:
+  Use this to listen for an event emitted in the deployed smart contracts.
+
+  ```ts
+  useScaffoldEventSubscriber({
+    contractName: "YourContract",
+    eventName: "GreetingChange",
+    //parameters that the event emits
+    //event GreetingChange(address greetingSetter, string newGreeting, bool premium, uint256 value);
+    listener: (greetingSetter, newGreeting, premium, value) => {
+      console.log(greetingSetter, newGreeting, premium, value);
+    },
+  });
+  ```
+
+  **useScaffoldEventHistory**:
+  Use this hook to read events from a deployed contract
+
+  ```ts
+  const {
+    data: events,
+    isLoading: isLoadingEvents,
+    error: errorReadingEvents,
+    } = useScaffoldEventHistory({
+    contractName: "YourContract",
+    eventName: "GreetingChange",
+    fromBlock: //block number here,
+    blockData: true,
```

Comment:
You can add the other optional parameters here and let the user know what are the mandatory parameters.

Reply:
I think Typescript will assist with that. the main aim of the doc is to show how to use it with its required parameters